### ISSUE TITLE
Job runtime is showing as negative

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/bespin-workflows#176]
on bespin-dev.gcb.duke.edu job 67 is showing with a negative runtime:

![screen shot 2018-11-19 at 10 59 58 am](https://user-images.githubusercontent.com/2752599/48718947-5b36e900-ebea-11e8-9909-79b852274a4e.png)

```
{
    "vm_hours": -0.09526106527777778,
    "cpu_hours": -3.048354088888889
}
```

**Comment by johnbradley** [comment]
I compared the output from the `date` command on the dev web and database servers/containers.
The web container is `EST` while the web server and database server/container are `UTC`.

**Comment by johnbradley** [comment]
I created a new job and was unable to see the error. The difference in timezone between the web container and the db container doesn't seem to make a difference in the calculation.
The database is recording the timezone and the django.util.timezone.now() includes the timezone.
The job activity for job 67 looks like a typical job (ie no cancel/restarts):
```
JobActivity - pk: 513 job.pk: 67 state: 'A' step: '' created: '2018-11-19 13:58:27.722225+00:00'
JobActivity - pk: 514 job.pk: 67 state: 'S' step: '' created: '2018-11-19 13:58:44.302474+00:00'
JobActivity - pk: 515 job.pk: 67 state: 'R' step: '' created: '2018-11-19 13:58:44.978109+00:00'
JobActivity - pk: 516 job.pk: 67 state: 'R' step: 'V' created: '2018-11-19 13:58:45.427494+00:00'
JobActivity - pk: 517 job.pk: 67 state: 'R' step: 'S' created: '2018-11-19 14:02:17.606286+00:00'
JobActivity - pk: 518 job.pk: 67 state: 'R' step: 'R' created: '2018-11-19 14:04:28.367329+00:00'
JobActivity - pk: 519 job.pk: 67 state: 'E' step: 'R' created: '2018-11-19 17:27:18.844005+00:00'
JobActivity - pk: 521 job.pk: 67 state: 'D' step: 'R' created: '2018-11-21 14:53:41.900374+00:00'
```
Using the django shell I calculated `vm_hours` for this job and it was positive.
Daylight savings change around this time was from November 4 so it should not have been a factor here.

**Comment by dleehr** [comment]
I thing the issue here is that the relationship does not specify any ordering for the JobActivity objects it [fetches](https://github.com/Duke-GCB/bespin-api/blob/029848f0aa6b9924137953b4d133aba8adae56f9/data/jobusage.py#L32) before calculating the time between them. So the database could easily return later objects before earlier ones, and inconsistently.

I confirmed too in the django shell that the `vm_hours` for this job is positive right now, but the database tables were obviously in a different state when I reported the issue, so that could be enough to cause a different order, couldn't it?

```
>>> from data.models import Job
>>> from data.jobusage import JobUsage
>>> j = Job.objects.get(id=67)
>>> u = JobUsage(j)
>>> u.vm_hours
3.4170104775
>>> u.cpu_hours
109.34433528
```

When asking Django for the underlying SQL query shows no `ORDER BY` clause:

```
>>> print(j.job_activities.all().query)
SELECT "data_jobactivity"."id", "data_jobactivity"."job_id", "data_jobactivity"."created", "data_jobactivity"."state", "data_jobactivity"."step" FROM "data_jobactivity" WHERE "data_jobactivity"."job_id" = 67
```
